seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java

```diff
@@ -224,8 +224,16 @@ public LinkedHashMap<String, List<String>> generatorPartitionDir(SeaTunnelRow se
     public final String generateFileName(String transactionId) {
         String fileNameExpression = fileSinkConfig.getFileNameExpression();
         FileFormat fileFormat = fileSinkConfig.getFileFormat();
-        String suffix = fileFormat.getSuffix();
-        suffix = compressFormat.getCompressCodec() + suffix;
+        String suffix;
+        if (StringUtils.isNotEmpty(fileSinkConfig.getFilenameExtension())) {
+            suffix =
+                    fileSinkConfig.getFilenameExtension().startsWith(".")
+                            ? fileSinkConfig.getFilenameExtension()
+                            : "." + fileSinkConfig.getFilenameExtension();
```

Comment:
`.`  may be widely used in this project, should we extract to the constant instead of using the magic value?

Reply:
I'm not sure if this is appropriate, as the `.` is pretty common. Not a specific text, maybe you could start a discussion on the mailing list. See what others think? It makes sense to me to change it or not. PS: If we need change it, we can open another PR.